.github/workflows/publish-sphinx.yml

```diff
@@ -1,5 +1,10 @@
 name: Publish Sphinx Docs to GitHub Pages
-on: [push]
+
+on:
```

Comment:
This change is not necessary since documentation will not be published unless new tag is pushed to the repository: https://github.com/nexusformat/definitions/blob/af7ea8ba357cc0651eb435bf2a6236da84b2206e/.github/workflows/publish-sphinx.yml#L109

Reply:
What if someone pushes a tag to another branch?

Reply:
Great question.  Looks possible right now.  Tags are not branch based, so the tag is on a commit hash in the repository.  Don't think we want any tags on branch commits.  What happens to the tag if the branch is closed without a merge?

Reply:
Yes I agree one should never push a tag to another branch than the default branch. But who knows, at some point someone might make a mistake.

Reply:
We actually never do it this way. One could always make mistakes when pushing tags.

We normally require someone to run the CI on the master manually with a flag that allows deployment explicitly.

https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow

Reply:
Probably needs its own issue.  When I get time, I want to read this page: https://git-scm.com/book/en/v2/Git-Basics-Tagging

Reply:
Agreed, it's a different issue.

Reply:
https://github.com/nexusformat/definitions/issues/1029

Reply:
In other repos, I've used release branches (which, I guess should be protected too) and tag releases on them.